lib/mocha/expectation.rb

```diff
@@ -220,7 +220,8 @@ def at_most_once
     #   object.expected_method(17)
     #   # => verify fails
     def with(*expected_parameters, &matching_block)
-      @parameters_matcher = ParametersMatcher.new(expected_parameters, &matching_block)
+      parameters = Parameters.from_ruby2_keywords(expected_parameters)
+      @parameters_matcher = ParametersMatcher.new(parameters, matching_block)
```

Comment:
This doesn't quite work as I hoped since `expected_parameters` can be matchers too